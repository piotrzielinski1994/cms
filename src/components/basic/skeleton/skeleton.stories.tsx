import type { Meta, StoryObj } from '@storybook/react';
import { type ComponentProps } from 'react';
import { Skeleton as SkeletonComponent } from './skeleton';

type Args = ComponentProps<typeof SkeletonComponent>;

const meta = {
  component: SkeletonComponent,
  title: 'Basic/Skeleton',
} satisfies Meta<Args>;

const Render = (_args: Args) => {
  return <SkeletonComponent />;
};

const Skeleton: StoryObj<typeof SkeletonComponent> = { render: Render };

export { Skeleton };
export default meta;
