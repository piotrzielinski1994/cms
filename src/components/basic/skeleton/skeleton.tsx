import { cn } from '@/utils/tailwind';
import { HTMLAttributes } from 'react';

type SkeletonProps = Omit<HTMLAttributes<HTMLDivElement>, 'children'>;

const Skeleton = ({ className, ...props }: SkeletonProps) => {
  return (
    <div
      role="status"
      aria-busy={true}
      {...props}
      className={cn(
        'h-[4lh] bg-components-skeleton relative overflow-hidden',
        'before:absolute before:inset-0 before:-translate-x-full',
        'before:bg-gradient-to-r before:from-transparent before:via-background/40 before:to-transfrom-transparent',
        'before:animate-skeleton',
        className,
      )}
    />
  );
};

export { Skeleton };
