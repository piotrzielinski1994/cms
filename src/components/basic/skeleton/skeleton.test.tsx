import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import { Skeleton } from './skeleton';

describe('Skeleton', () => {
  it('should have no accessibility violations', async () => {
    const { container } = render(<Skeleton />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(<Skeleton />);
    expect(container).toMatchSnapshot();
  });
});
