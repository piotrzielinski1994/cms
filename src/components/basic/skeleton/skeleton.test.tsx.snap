// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > should match the snapshot 1`] = `
<div>
  <div
    aria-busy="true"
    class="h-[4lh] bg-components-skeleton relative overflow-hidden before:absolute before:inset-0 before:-translate-x-full before:bg-gradient-to-r before:from-transparent before:via-background/40 before:to-transfrom-transparent before:animate-skeleton"
    role="status"
  />
</div>
`;
